Mathlib/Analysis/SpecialFunctions/Log/ENNRealLogExp.lean

```diff
@@ -0,0 +1,186 @@
+/-
+Copyright (c) 2024 Rémy Degenne. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damien Thomine, Pietro Monticone, Rémy Degenne, Lorenzo Luccioli
+-/
+import Mathlib.Analysis.SpecialFunctions.Log.ERealExp
+import Mathlib.Analysis.SpecialFunctions.Log.ENNRealLog
+import Mathlib.MeasureTheory.Constructions.BorelSpace.Basic
+import Mathlib.Topology.MetricSpace.Polish
+
+/-!
+# Properties of the extended logarithm and exponential
+
+We prove that `log` and `exp` define order isomorphisms between `ℝ≥0∞` and `EReal`.
+## Main Definitions
+- `ENNReal.logOrderIso`: The order isomorphism between `ℝ≥0∞` and `EReal` defined by `log`
+and `exp`.
+- `ENNReal.logHomeomorph`: `log` as a homeomorphism.
+- `EReal.expHomeomorph`: `exp` as a homeomorphism.
+
+## Main Results
+- `ENNReal.log_exp`, `ENNReal.exp_log`: `log` and `exp` are inverses of each other.
+- `ENNReal.exp_nmul`, `ENNReal.exp_mul`: `exp` satisfies the identities `exp (n * x) = (exp x) ^ n`
+and `exp (x * y) = (exp x) ^ y`.
+- `instMetrizableSpaceEReal`: `EReal` is a metrizable space.
+
+## Tags
+ENNReal, EReal, logarithm, exponential
+-/
+
+open EReal ENNReal
+section LogExp
+
+@[simp] lemma EReal.log_exp (x : EReal) : log (exp x) = x := by
+  induction x
+  · simp
+  · rw [exp_coe, log_ofReal, if_neg (not_le.mpr (Real.exp_pos _)), Real.log_exp]
+  · simp
+
+@[simp] lemma ENNReal.exp_log (x : ℝ≥0∞) : exp (log x) = x := by
+  by_cases hx_top : x = ∞
+  · simp [hx_top]
+  by_cases hx_zero : x = 0
+  · simp [hx_zero]
+  have hx_pos : 0 < x.toReal := ENNReal.toReal_pos hx_zero hx_top
+  rw [← ENNReal.ofReal_toReal hx_top, log_ofReal_of_pos hx_pos, exp_coe, Real.exp_log hx_pos]
+
+end LogExp
+
+section Log
+namespace ENNReal
+
+@[simp]
+lemma log_lt_log_iff {x y : ℝ≥0∞} : log x < log y ↔ x < y := by
+  refine ⟨fun h ↦ ?_, fun h ↦ log_strictMono h⟩
+  rw [← exp_log x, ← exp_log y]
+  exact exp_strictMono h
```

Comment:
This is true for any strict mono function, so it has to be a one-liner. Indeed:
```suggestion
lemma log_lt_log_iff {x y : ℝ≥0∞} : log x < log y ↔ x < y := log_strictMono.lt_iff_lt
```

But then this lemma and the lemmas that follow don't use the exponential, and should probably go in the file about log.